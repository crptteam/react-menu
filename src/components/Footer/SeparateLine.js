import React from 'react';
import SeparateLineWrapper from '../../styled/SeparateLineWrapper';


const SeparateLine = () => (<SeparateLineWrapper />);
SeparateLine.displayName = 'SeparateLine';

export default SeparateLine;
