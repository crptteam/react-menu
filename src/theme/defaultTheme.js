export default {
  Menu: {
    common: {
      width: {
        full: '241px',
        fold: '78px',
      },
      backgroundColor: '#FFFFFF',
      borderColor: 'rgba(0, 0, 0, 0.25)',
      borderWidth: '1px',
      shadowX: '0px',
      shadowY: '2px',
      shadowColor: 'rgba(0, 0, 0, 0.25)',
    },
    header: {
      height: '200px',
      alignItems: 'center',
      marginTop: '10px',
      marginBottom: '10px',
    },
    item: {
      width: '80%',
      height: '52px',
      borderRadius: '4px',
      marginTop: '10px',
      marginBottom: '10px',
      hoverBorderWidth: '1px',
      hoverBorderColor: '#F8EC31',
      shadowX: '0px',
      shadowY: '2px',
      shadowColor: 'rgba(0, 0, 0, 0.25)',
      justifyContentOnFull: 'flex-start',
      justifyContentOnFold: 'center',
      isSelectedColor: '#F8EC31',
      activeStyle: {
        isSelectedColor: '#F8EC31',
      },
      textColor: 'black',
    },
    fullModeButton: {
      height: '80px',
    },
    separator: {
      width: '80%',
      height: '1px',
      color: 'rgba(0, 0, 0, 0.25)',
    },
    footer: {
      height: '240px',
      alignItems: 'center',
      marginTop: '10px',
      marginBottom: '10px',
    },
  },
};
