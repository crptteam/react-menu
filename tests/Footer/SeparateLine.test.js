import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import defaultTheme from '../../src/theme/defaultTheme';

import SeparateLine from '../../src/components/Footer/SeparateLine';

describe('Footer/SeparateLine', () => {
  it('should renders without problem', () => {
    const component = mount(
      <ThemeProvider theme={defaultTheme}>
        <SeparateLine />
      </ThemeProvider>
    );
    expect(component).toMatchSnapshot();
  });
});
