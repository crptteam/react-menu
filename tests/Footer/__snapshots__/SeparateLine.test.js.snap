// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer/SeparateLine should renders without problem 1`] = `
<ThemeProvider
  theme={
    Object {
      "Menu": Object {
        "common": Object {
          "backgroundColor": "#FFFFFF",
          "borderColor": "rgba(0, 0, 0, 0.25)",
          "borderWidth": "1px",
          "shadowColor": "rgba(0, 0, 0, 0.25)",
          "shadowX": "0px",
          "shadowY": "2px",
          "width": Object {
            "fold": "78px",
            "full": "241px",
          },
        },
        "footer": Object {
          "alignItems": "center",
          "height": "240px",
          "marginBottom": "10px",
          "marginTop": "10px",
        },
        "fullModeButton": Object {
          "height": "80px",
        },
        "header": Object {
          "alignItems": "center",
          "height": "200px",
          "marginBottom": "10px",
          "marginTop": "10px",
        },
        "item": Object {
          "borderRadius": "4px",
          "height": "52px",
          "hoverBorderColor": "#F8EC31",
          "hoverBorderWidth": "1px",
          "isSelectedColor": "#F8EC31",
          "justifyContentOnFold": "center",
          "justifyContentOnFull": "flex-start",
          "marginBottom": "10px",
          "marginTop": "10px",
          "shadowColor": "rgba(0, 0, 0, 0.25)",
          "shadowX": "0px",
          "shadowY": "2px",
          "width": "80%",
        },
        "separator": Object {
          "color": "rgba(0, 0, 0, 0.25)",
          "height": "1px",
          "width": "80%",
        },
      },
    }
  }
>
  <SeparateLine>
    <WithTheme(SeparateLineWrapper)>
      <SeparateLineWrapper
        theme={
          Object {
            "Menu": Object {
              "common": Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "rgba(0, 0, 0, 0.25)",
                "borderWidth": "1px",
                "shadowColor": "rgba(0, 0, 0, 0.25)",
                "shadowX": "0px",
                "shadowY": "2px",
                "width": Object {
                  "fold": "78px",
                  "full": "241px",
                },
              },
              "footer": Object {
                "alignItems": "center",
                "height": "240px",
                "marginBottom": "10px",
                "marginTop": "10px",
              },
              "fullModeButton": Object {
                "height": "80px",
              },
              "header": Object {
                "alignItems": "center",
                "height": "200px",
                "marginBottom": "10px",
                "marginTop": "10px",
              },
              "item": Object {
                "borderRadius": "4px",
                "height": "52px",
                "hoverBorderColor": "#F8EC31",
                "hoverBorderWidth": "1px",
                "isSelectedColor": "#F8EC31",
                "justifyContentOnFold": "center",
                "justifyContentOnFull": "flex-start",
                "marginBottom": "10px",
                "marginTop": "10px",
                "shadowColor": "rgba(0, 0, 0, 0.25)",
                "shadowX": "0px",
                "shadowY": "2px",
                "width": "80%",
              },
              "separator": Object {
                "color": "rgba(0, 0, 0, 0.25)",
                "height": "1px",
                "width": "80%",
              },
            },
          }
        }
      >
        <SeparateLineWrapper__SeparateLineDiv
          separatorTheme={
            Object {
              "color": "rgba(0, 0, 0, 0.25)",
              "height": "1px",
              "width": "80%",
            }
          }
          theme={
            Object {
              "Menu": Object {
                "common": Object {
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "rgba(0, 0, 0, 0.25)",
                  "borderWidth": "1px",
                  "shadowColor": "rgba(0, 0, 0, 0.25)",
                  "shadowX": "0px",
                  "shadowY": "2px",
                  "width": Object {
                    "fold": "78px",
                    "full": "241px",
                  },
                },
                "footer": Object {
                  "alignItems": "center",
                  "height": "240px",
                  "marginBottom": "10px",
                  "marginTop": "10px",
                },
                "fullModeButton": Object {
                  "height": "80px",
                },
                "header": Object {
                  "alignItems": "center",
                  "height": "200px",
                  "marginBottom": "10px",
                  "marginTop": "10px",
                },
                "item": Object {
                  "borderRadius": "4px",
                  "height": "52px",
                  "hoverBorderColor": "#F8EC31",
                  "hoverBorderWidth": "1px",
                  "isSelectedColor": "#F8EC31",
                  "justifyContentOnFold": "center",
                  "justifyContentOnFull": "flex-start",
                  "marginBottom": "10px",
                  "marginTop": "10px",
                  "shadowColor": "rgba(0, 0, 0, 0.25)",
                  "shadowX": "0px",
                  "shadowY": "2px",
                  "width": "80%",
                },
                "separator": Object {
                  "color": "rgba(0, 0, 0, 0.25)",
                  "height": "1px",
                  "width": "80%",
                },
              },
            }
          }
        >
          <div
            className="SeparateLineWrapper__SeparateLineDiv-hyoUrY fZONmc"
          />
        </SeparateLineWrapper__SeparateLineDiv>
      </SeparateLineWrapper>
    </WithTheme(SeparateLineWrapper)>
  </SeparateLine>
</ThemeProvider>
`;
